code/obj/machinery/food_and_drink/mixer.dm

```diff
@@ -239,13 +241,31 @@ TYPEINFO(/obj/machinery/mixer)
 		UnsubscribeProcess()
 		return
 
+	power_change()
+		. = ..()
+		src.UpdateIcon()
+
+	disposing()
+		. = ..()
+		src.ClearSpecificOverlays(0)
+
 	update_icon()
 		if (!src || !istype(src))
 			return
 
-		if (src.working != 0)
-			src.icon_state = "blender_on"
+		if (length(src.contents) > 0)
+			src.icon_state = "blender_filled"
+		else
+			src.icon_state = "blender_empty"
+
+		src.ClearSpecificOverlays(1)
+		var/powered = src.status & NOPOWER
+
+		if (powered)
+			src.UpdateOverlays(image(src.icon, "blender_off"), "blender_off")
+		else if (!powered && !src.working)
+			src.UpdateOverlays(image(src.icon, "blender_powered"), "blender_powered")
 		else
-			src.icon_state = "blender"
+			src.UpdateOverlays(image(src.icon, "blender_working"), "blender_working")
```

Comment:
aren't these all unique? it should be like `blender_light` - then you don't need to clear stuff all the time

Reply:
I misunderstood how the function worked, makes more sense this way and implemented now. Thanks!